src/rendering/vcTileRenderer.cpp

```diff
@@ -214,27 +228,93 @@ uint32_t vcTileRenderer_LoadThread(void *pThreadData)
       int channelCount = 0;
       uint8_t *pData = nullptr;
 
-      UD_ERROR_CHECK(udFile_Load(pTileURL, &pFileData, &fileLen));
-      UD_ERROR_IF(fileLen == 0, udR_InternalError);
+      char localFileName[vcMaxPathLength] = {};
+      char serverAddress[vcMaxPathLength] = {};
+      bool downloadingFromServer = true;
 
-      // Node has been invalidated since download started
-      if (!pBestNode->touched)
+      vcQuadTreeNode *pBestNode = pCache->tileLoadList[best];
+      pCache->tileLoadList.RemoveSwapLast(best);
+
+      // TODO: Processing a dem request has been stuck into this block. This has been rushed
+      // and will be looked at later.
+
+      // process either a colour request, or a DEM request
+      if (pBestNode->renderInfo.loadStatus == vcNodeRenderInfo::vcTLS_InQueue)
       {
-        // TODO: Put into LRU texture cache (but for now just throw it out)
-        pBestNode->renderInfo.loadStatus = vcNodeRenderInfo::vcTLS_None;
+        // do colour request
+        pBestNode->renderInfo.loadStatus = vcNodeRenderInfo::vcTLS_Downloading;
+
+        udSprintf(localFileName, "%s/%s/%d/%d/%d.%s", pRenderer->pSettings->cacheAssetPath, udUUID_GetAsString(pRenderer->pSettings->maptiles.tileServerAddressUUID), pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y, pRenderer->pSettings->maptiles.tileServerExtension);
+        udSprintf(serverAddress, "%s/%d/%d/%d.%s", pRenderer->pSettings->maptiles.tileServerAddress, pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y, pRenderer->pSettings->maptiles.tileServerExtension);
+        udReleaseMutex(pCache->pMutex);
+
+        char *pTileURL = serverAddress;
+
+        if (udFileExists(localFileName) == udR_Success)
```

Comment:
You should flip this around. `pTileURL` is set to `localFileName` and then this checks if `pTileURL` exists, otherwise it sets `pTileURL` to `serverAddress`. Additionally, you can avoid the `udSprintf` for the server address when it there's a local file.

Reply:
Done.

I'm going to leave the `udSprintf` doing extra work because `localFileName` and `serverAddress` should be created thread safe.